src/charm.py

```diff
@@ -371,6 +371,11 @@ def _reconcile(self):
 
         self.unit.status = ActiveStatus()
 
+        # Mandatory relation pairs
+        missing_relations = integrations.get_missing_mandatory_relations(self)
+        if missing_relations:
+            self.unit.status = BlockedStatus(missing_relations)
```

Comment:
Small suggestion to reduce lines of code :P

```suggestion
        if missing_relations := integrations.get_missing_mandatory_relations(self):
            self.unit.status = BlockedStatus(missing_relations)
```